src/state/apis/zapper/zapperApi.ts

```diff
@@ -79,37 +82,66 @@ export const zapperApi = createApi({
           },
         })
 
-        const data = zapperV2AppTokensData.map(appTokenData =>
-          toAssetId({
-            chainId: zapperNetworkToChainId(appTokenData.network)!,
-            assetNamespace: 'erc20', // TODO: bep20
-            assetReference: appTokenData.address,
-          }),
+        const data = zapperV2AppTokensData.reduce<GetZapperAppBalancesOutput>(
+          (acc, appTokenData) => {
+            const assetId = toAssetId({
+              chainId: zapperNetworkToChainId(appTokenData.network),
+              assetNamespace: 'erc20', // TODO: bep20
+              assetReference: appTokenData.address,
+            })
+
+            acc[assetId] = appTokenData
+            return acc
+          },
+          {},
         )
 
+        const assets = selectAssets(getState() as any)
```

Comment:
```suggestion
        const assets = selectAssets(getState() as ReduxState)
```

i think you can cast here without circular dep

Reply:
Very noice - 588bdead8